### PR TITLE
Prepare deployment configuration for k8s with Helm

### PR DESCRIPTION
**Pull request by petertrr**
* Add Helm chart `save-cloud` into the repo
* For each spring-boot service there are: Deployment, Service and ConfigMap; some also use Secrets; Pods are configured to expose liveness endpoints in actuator
* PersistentVolumeClaim for fs-storage (binaries, images etc.)
* Entrypoint into the application is `gateway-service`, which can be configured via `values.yaml` to serve as [loadbalancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) depending on a specific cloud provider
* Add Service for the external database
* Add `dependencies.yaml` for third-party services. Currently contains grafana.
* Add github workflow to test the chart using `chart-testing`
* Add renovate bot config for Helm
* Support overriding backend and orchestrator URLs in agent.properties based on orchestrator properties

### Current limitations:
* Currently these manual steps are required for node configuration:
  * Install gvisor
  * Enable HTTP API for Docker daemon
  * Create directory for repositories with correct permissions (e.g. `chown -R 1000:1000 /tmp/save/repositories`)
* No support for database migrations
* save-agents are deployed into plain Docker environment

### "Backlog" for further PRs:
* Database migrations: init containers: `alpine/git` to clone repo, image with preinstalled gradle to run migrations with gradle or use `liquibase/liquibase` to avoid slow downloading of all project dependencies. Links: https://www.liquibase.com/blog/using-liquibase-in-kubernetes , https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/
* Automatic node configuration. Add DaemonSet to enable docker HTTP API on the node and to [install gvisor](https://www.danielstechblog.io/running-gvisor-on-azure-kubernetes-service-for-sandboxing-containers/) runsc runtime on a new node automatically.
* Orchestrator: deploy agents (`save-execution` containers) into k8s so that they can access orchestrator via k8s DNS. For now orchestrator can use docker daemon HTTP API on the same host, and agents can use fixed ClusterIP of backend and orchestrator services
* Enhancement: different values per environment (staging/prod)
* Add Prometheus and Loki into the chart
* `save-fs-storage`: support proper distributed storage, or even better move this data to S3-compatible storage
* Chart tests: https://v2.helm.sh/docs/developing_charts/#chart-tests
* Scalability: 
  * Staying logged in with several instances of api-gateway: [sticky session](https://kubernetes.github.io/ingress-nginx/examples/affinity/cookie/)
  * Database locking between instances of backend (if different instances would edit same data)
*  `hostPath` directories cannot be mounted with permissions change. Workaround: use DaemonSet or InitContainers or something similar to prepare directory on the node. For a time can be done manually.
* Uncouple frontend from backend; introduce a separate Pod that would run e.g. nginx to serve static content
  * Uncouple scheduled tasks from backend using CronJobs

### CODE REVIEW

File: save-cloud-charts/save-cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Helm chart for save-cloud
```
Comment (akuleshov7) [inline save-cloud-charts/save-cloud/README.md:1]: pls add links for a local installation of Kuber

File: save-cloud-charts/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    commandLine("helm", "repo")
+    args("add", "grafana", "https://grafana.github.io/helm-charts")
+    inputs.file("save-cloud/requirements.yaml")
+}
+
+val chartVersion = "0.1.0"  // same as in `Chart.yaml`
+val chartArchive = "$buildDir/save-cloud-$chartVersion.tgz"
+tasks.register<Exec>("packageChart") {
+    inputs.dir("save-cloud")
+    commandLine("helm", "package")
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-cloud-charts/build.gradle.kts:22]: I believe gradle logs them with `--info` already, do we need to duplicate it?

File: save-cloud-charts/save-cloud/README.md (hunk 1 above)
Comment (petertrr) [reply save-cloud-charts/save-cloud/README.md:1]: Added

**Review by akuleshov7 — APPROVED** [review]

File: save-cloud-charts/build.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (petertrr) [reply save-cloud-charts/build.gradle.kts:22]: For example, `gradlew :save-cloud-charts:setupChartRepositories`
```
Starting process 'command 'helm''. Working directory: D:\projects\save-cloud\save-cloud-charts Command: helm repo add grafana https://grafana.github.io/helm-charts
```

**Pull request merged**